### ISSUE TITLE
Support EC2 Create Tags Resource

### ISSUE THREAD

**Issue by e-moshaya** [terraform-providers/terraform-provider-aws#12500]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

aws_spot_instance_request supports creating tags to the spot instance request but not the instance or the volumes. The tags do not get propagated as first thought. [Amazon suggests](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html) tagging the instances manually using the cli `aws ec2 create-tags --resources sir-08b93456 i-1234567890abcdef0 --tags Key=purpose,Value=test`

It would be nice to introduce a new resource to create or overwrite tags on the fly in a similar fashion

### New or Affected Resource(s)

* aws_ec2_create_tags

### Potential Terraform Configuration

```
resource "aws_ec2_create_tags" "spot_instance" {
  resources           = ["sir-08b93456", "i-1234567890abcdef0"]
  tags = {
    Name = "spot-instance"
  }
}
```

### References

* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html
* https://docs.aws.amazon.com/cli/latest/reference/ec2/create-tags.html

**Comment by ewbankkit** [comment]
Similar:
* https://github.com/terraform-providers/terraform-provider-aws/issues/9061
* https://github.com/terraform-providers/terraform-provider-aws/issues/11634

Related: https://github.com/terraform-providers/terraform-provider-aws/pull/8457.

**Comment by bflad** [comment]
A new `aws_ec2_tag` resource for managing individual EC2 resource tags has been merged and will release with version 2.67.0 of the Terraform AWS Provider, later next week. This resource should only be used in cases where EC2 resources are created outside Terraform (e.g. AMIs), being shared via Resource Access Manager (RAM), or implicitly created by other means (e.g. Transit Gateway VPN Attachments).

```hcl
# Example configuration in Terraform 0.12 and later syntax
resource "aws_ec2_transit_gateway" "example" {}

resource "aws_customer_gateway" "example" {
  bgp_asn    = 65000
  ip_address = "172.0.0.1"
  type       = "ipsec.1"
}

resource "aws_vpn_connection" "example" {
  customer_gateway_id = aws_customer_gateway.example.id
  transit_gateway_id  = aws_ec2_transit_gateway.example.id
  type                = aws_customer_gateway.example.type
}

resource "aws_ec2_tag" "example" {
  resource_id = aws_vpn_connection.example.transit_gateway_attachment_id
  key         = "Name"
  value       = "Hello World"
}
```

As with any Terraform 0.12.6 or later configuration, this resource can be combined with [`for_each` support](https://www.terraform.io/docs/configuration/resources.html#for_each-multiple-resource-instances-defined-by-a-map-or-set-of-strings) to manage multiple resource tags, if necessary.

Thanks to @joestump and others who made the implementation possible. 👍
